indiserver/CommandLineArgs.hpp

```diff
@@ -0,0 +1,38 @@
+/* INDI Server for protocol version 1.7.
+ * Copyright (C) 2007 Elwood C. Downey <ecdowney@clearskyinstitute.com>
+                 2013 Jasem Mutlaq <mutlaqja@ikarustech.com>
+                 2022 Ludovic Pollet
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+#pragma once
+#include "Constants.hpp"
+
+#include <string>
+
+class Fifo;
+
+struct CommandLineArgs
+{
+    size_t verbosity{0};
+    unsigned int maxStreamSizeMB{indiserver::constants::defaultMaxStreamSizeMB};
+    unsigned int maxQueueSizeMB{indiserver::constants::defaultMaxQueueSizeMB};
+    char *loggingDir{nullptr};
+    int maxRestartAttempts{indiserver::constants::defaultMaximumRestarts};
+    Fifo* fifoHandle{nullptr};
```

Comment:
This is not really a command line argument,  but rather a service instance built from it. (As Unix server, tcp server,... ) . Could you decouple the argument from the implementation ?

Reply:
Yeah I wasn't really sure if it belonged in here or separate. I'll move it out.

Reply:
Moved it out with 8e33aff